File: lista 15questoes/15questoes em golang/att14.go
package main

import "sync"

type Configuracao struct{}

var instancia *Configuracao
var once sync.Once

func GetConfiguracao() *Configuracao {
    once.Do(func() {
        instancia = &Configuracao{}
    })
    return instancia
}

config1 := GetConfiguracao()
config2 := GetConfiguracao()
fmt.Println(config1 == config2)  // true

File: lista 15questoes/15 questoes em java/att14.java
class Configuracao {
    private static Configuracao instancia;

    private Configuracao() {}

    public static Configuracao getInstancia() {
        if (instancia == null) {
            instancia = new Configuracao();
        }
        return instancia;
    }
}

Configuracao config1 = Configuracao.getInstancia();
Configuracao config2 = Configuracao.getInstancia();
System.out.println(config1 == config2);  // true
